common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -117,6 +141,27 @@ def build_mention_nodes(text)
           nodes
         end
 
+        def build_team_mention_nodes(text)
+          nodes = []
+
+          scan = StringScanner.new(text)
+          until scan.eos?
+            line = scan.scan_until(TEAM_MENTION_REGEX) ||
+                   scan.scan_until(EOS_REGEX)
+            line_match = line.match(TEAM_MENTION_REGEX)
+            mention = line_match&.to_s
+            text_node = CommonMarker::Node.new(:text)
+            text_node.string_content = line
+            next unless mention
+
+            text_node.string_content = line_match.pre_match
```

Comment:
Looks like the text node isn't used unless there's a mention? Would something like this work?

```suggestion
            next unless mention

            text_node = CommonMarker::Node.new(:text)
            text_node.string_content = line_match.pre_match
```

Reply:
Or should we also do what we do above in `build_mention_nodes` and always add the text_node if there is no mention?

e.g.

```
if mention
  text_node.string_content = line_match.pre_match
  nodes << text_node
  nodes << build_mention_link_text_nodes(mention.to_s).first
else
  text_node.string_content = line
  nodes << text_node
end
```

I can't actually remember what that case does for regular mentions but maybe we have some weird test cases covering it?

Reply:
@jurre just thought of above, might not be needed but maybe worth checking the existing test suite to see when we end up in the `else` for mentions?

Reply:
> Or should we also do what we do above in build_mention_nodes and always add the text_node if there is no mention?

Ah yes we do, otherwise we'll end up dropping lines without mentions. Will add a test to cover this

Reply:
Ah, no we don't. The reason that case is there for regualar mentions is because @feelepxyz/ doesn't need to be sanitized, that's the `&& !mention.end_with?("/")` case

Reply:
Ah gotcha! Thanks for verifying 👍